safe_control_gym/controllers/mpc/mpc.py

```diff
@@ -291,14 +348,39 @@ def select_action(self,
             opti.set_initial(x_var, x_guess)
             opti.set_initial(u_var, u_guess)
         # Solve the optimization problem.
-        sol = opti.solve()
-        x_val, u_val = sol.value(x_var), sol.value(u_var)
+        try:
+            sol = opti.solve()
+            x_val, u_val = sol.value(x_var), sol.value(u_var)
+        except RuntimeError:
+            print(colored('Infeasible MPC Problem', 'red'))
+            if self.solver == 'ipopt':
+                x_val, u_val = opti.debug.value(x_var), opti.debug.value(u_var)
+            elif self.solver == 'qrsqp':
+                return_status = opti.return_status()
+                print(f'Optimization failed with status: {return_status}')
+                if return_status == 'unknown':
+                    # self.terminate_loop = True
+                    u_val = self.u_prev
+                    x_val = self.x_prev
+                    if u_val is None:
+                        print('[WARN]: MPC Infeasible first step.')
+                        u_val = u_guess
+                        x_val = x_guess
+                elif return_status == 'Maximum_Iterations_Exceeded':
```

Comment:
these two if statements lead to the same result

Reply:
This is fixed in the latest commit.